WcaOnRails/app/models/registration.rb

```diff
@@ -24,12 +27,30 @@ class Registration < ActiveRecord::Base
     end
   end
 
-  def pending?
-    accepted_at.nil?
+  def deleted?
+    !deleted_at.nil?
   end
 
   def accepted?
-    !pending?
+    !deleted? && !accepted_at.nil?
+  end
+
+  def pending?
+    !accepted? && !deleted?
+  end
+
+  def checked_status
```

Comment:
I'm okay with doing this this way for this PR, but could we make an issue to add an explicit `status` field to Registration? I think that would eventually make this cleaner.

Could we also add a validation that says a registration cannot be both accepted and deleted simultaneously?